control-plane/agents/src/bin/ha/cluster/server.rs

```diff
@@ -51,17 +64,47 @@ impl ClusterAgentOperations for ClusterAgentSvc {
             ));
         }
 
+        let ep: SocketAddr = request.endpoint().parse().unwrap();
+
+        self.nodes.register_node(request.node(), ep).await;
         tracing::trace!(agent = request.node(), "node successfully registered");
         Ok(())
     }
 
+    #[tracing::instrument(level = "info", skip(self), err, fields(node.name = %request.node()))]
     async fn report_failed_nvme_paths(
         &self,
-        _request: &dyn ReportFailedPathsInfo,
+        request: &dyn ReportFailedPathsInfo,
         _context: Option<Context>,
     ) -> Result<(), ReplyError> {
-        Err(ReplyError::unimplemented(
-            "NVMe path reporting is not yet implemented".to_string(),
-        ))
+        let mut v: Vec<(String, String)> = Vec::new();
+
+        for x in request.failed_paths().into_iter() {
```

Comment:
shouldn't we use a single report failed path at once? I seem to remember discussing this before @mtzaurus